src/main/java/com/hazelcast/commandline/member/MemberCommandLine.java

```diff
@@ -0,0 +1,341 @@
+
+    private void removeProcess(String processUniqueId) throws IOException, ClassNotFoundException {
+        Map<String, HazelcastProcess> processMap = getProcessMap();
+        if (processMap == null || !processMap.containsKey(processUniqueId)){
+            println("No process found with pid: " + processUniqueId);
+            return;
+        }
+        processMap.remove(processUniqueId);
+        updateFile(processMap);
+    }
+
+    public Map<String, HazelcastProcess> getProcessMap() throws ClassNotFoundException {
+        FileInputStream fileInputStream;
+        Map<String, HazelcastProcess> processes = new HashMap<>();
+        try {
+            Path path = FileSystems.getDefault().getPath(instancesFilePath);
+            if (!Files.exists(path)){
+                Files.createFile(path);
+            }
+            fileInputStream = new FileInputStream(instancesFilePath);
+            if (Files.size(path) == 0){
+                return processes;
+            }
+            ObjectInputStream input = new ObjectInputStream(fileInputStream);
+            processes = (Map<String, HazelcastProcess>) input.readObject();
+            input.close();
+        } catch (IOException e) {
+            throw new HazelcastException("Error when reading from file.", e);
+        }
+        return processes;
+    }
+
+    private Integer buildJavaProcess(Class aClass, List<String> parameters, boolean foreground, String additionalClassPath)
+            throws IOException, InterruptedException {
+        List<String> commandList = new ArrayList<>();
+        String classpath = System.getProperty("java.class.path");
+        if (!isNullOrEmpty(additionalClassPath)){
+            classpath += CLASSPATH_SEPARATOR + additionalClassPath;
+        }
+        String path = System.getProperty("java.home")
+                + SEPARATOR + "bin" + SEPARATOR + "java";
+        commandList.add(path);
+        commandList.add("-cp");
+        commandList.add(classpath);
+        commandList.addAll(parameters);
+        commandList.add(aClass.getName());
+        ProcessBuilder processBuilder = new ProcessBuilder(commandList);
+        processBuilder.redirectErrorStream(true);
+        if (foreground) {
+            processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+        }
+        Process process = processBuilder.start();
+        if (foreground) {
+            process.waitFor();
+        }
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        processOutput = bufferedReader.lines();
+        return getPid(process);
+    }
+
+    private void saveProcess(HazelcastProcess process) throws IOException, ClassNotFoundException {
+        Map<String, HazelcastProcess> processMap = getProcessMap();
+        processMap.put(process.getProcessUniqueId(), process);
+        updateFile(processMap);
+    }
+
+    private void updateFile(Map<String, HazelcastProcess> processMap) throws IOException {
```

Comment:
1. Yes, we should discuss about this issue. If you are ok, we can move as it is for now and create a specific GH issue for it to discuss the details.
2. I didn't prefer human-readable format in order not to make it available easy changes. Otherwise, it can cause a consistency issue easily.